File: hw/vizcaino/u3/hw24FactoryMethod/hw24FactoryMethodJavaScript/guiFactory.js
class GUIFactory {
    constructor() {
        if (this.constructor === GUIFactory) {
            throw new Error("Cannot instantiate abstract class GUIFactory");
        }
    }

    static getFactory() {
        const sys = GUIFactory.readFromConfigFile("OS_TYPE");
        if (sys === 0) {
            const WinFactory = require('./winFactory');
            return new WinFactory();
        } else {
            const LinuxFactory = require('./linuxFactory');
            return new LinuxFactory();
        }
    }

    static readFromConfigFile(key) {
        return 0;
    }

    createButton() {
        throw new Error("Method 'createButton()' must be implemented.");
    }

    createMenu() {
        throw new Error("Method 'createMenu()' must be implemented.");
    }
}

module.exports = GUIFactory;

File: hw/nacato/u3/hw23AbstractFactory/hw23AbstractFactory.java/src/main/java/ec/edu/espe/AbstractFactory/model/GUIFactory.java

package ec.edu.espe.AbstractFactory.model;

/**
 *
 * @author Angie Ñacato, Error 404, @ESPE
 */
public abstract class GUIFactory {
    public static GUIFactory getFactory() {
        int sys = readFromConfigFile("OS_TYPE");
        if (sys == 0) {
            return new WinFactory();
        } else {
            return new LinuxFactory();
        }
    }

    private static int readFromConfigFile(String key) {
        String os = System.getProperty("os.name").toLowerCase();
        if (os.contains("win")) {
            return 0;
        } else {
            return 1;
        }
    }

    public abstract Button createButton();
    public abstract Menu createMenu();
}
